src/prefect/tasks/sftp/sftp.py

```diff
@@ -0,0 +1,280 @@
+import errno, time
+import os.path
+from paramiko import Transport, SFTPClient
+
+from prefect import Task
+from prefect.utilities.tasks import defaults_from_attrs
+
+
+class SftpDownload(Task):
+    """
+    Task for downloading files from an SFTP server.
+    Downloads remote file into sftp_downloads/ folder by default
+
+    Args:
+        - host (str): name of the host to use.
+        - username (str): username used to authenticate.
+        - password (str): password used to authenticate.
+        - port_number (int): the port number to connect to the server.
+        - remote_path (str): the remote sftp file path
+        - local_path (str): the local file path to download file to
+        - **kwargs (dict, optional): additional keyword arguments to pass to the Task constructor.
+
+    Raises:
+        - ValueError: if a required parameter is not supplied.
+        - ClientError: if exception occurs when connecting/downloading from the server.
+    """
+
+    def __init__(
+        self,
+        host: str = None,
+        username: str = None,
+        password: str = None,
+        port_number: int = None,
+        remote_path: str = None,
+        local_path: str = None,
+        **kwargs,
+    ):
+        self.host = host
+        self.username = username
+        self.password = password
+        self.port_number = port_number
+        self.remote_path = remote_path
+        self.local_path = local_path
+        super().__init__(**kwargs)
+
+    def _create_connection(self) -> None:
+        """
+        Initialise the connection with the SFTP server
+        :return: None
+        """
+        transport = Transport(sock=(self.host, self.port_number))
+        transport.connect(username=self.username, password=self.password)
+        self._connection = SFTPClient.from_transport(transport)
+        self.logger.info(f"connected to {self.host}, {self.port_number}")
+
+    def file_exists(self, remote_path: str) -> bool:
+        """
+        Checks if file exists in remote path or not.
+
+        Args:
+            - remote_path (str): checks if the specified remote
+            file path exists on the sftp server
+
+        """
+        res: bool = True
```

Comment:
My recommendation would be to default `res` to `False`. Then you can eliminate your `except` clause and avoid false positives.